Sources/Baggage/Baggage.swift

```diff
@@ -0,0 +1,237 @@
+    /// not being sure where to obtain a context from, or other framework limitations -- e.g. the outer framework not being
+    /// baggage context aware just yet.
+    public static var background: Baggage {
+        return Baggage()
+    }
+}
+
+extension Baggage {
+    /// A baggage intended as a placeholder until a real value can be passed through a function call.
+    ///
+    /// It should ONLY be used while prototyping or when the passing of the proper context is not yet possible,
+    /// e.g. because an external library did not pass it correctly and has to be fixed before the proper context
+    /// can be obtained where the TO-DO is currently used.
+    ///
+    /// ## Crashing on TO-DO context creation
+    /// You may set the `BAGGAGE_CRASH_TODOS` variable while compiling a project in order to make calls to this function crash
+    /// with a fatal error, indicating where a to-do baggage context was used. This comes in handy when wanting to ensure that
+    /// a project never ends up using with code initially was written as "was lazy, did not pass context", yet the
+    /// project requires context passing to be done correctly throughout the application. Similar checks can be performed
+    /// at compile time easily using linters (not yet implemented), since it is always valid enough to detect a to-do context
+    /// being passed as illegal and warn or error when spotted.
+    ///
+    /// ## Example
+    ///
+    ///     frameworkHandler { what in
+    ///         hello(who: "World", baggage: .TODO("The framework XYZ should be modified to pass us a context here, and we'd pass it along"))
+    ///     }
+    ///
+    /// - Parameters:
+    ///   - reason: Informational reason for developers, why a placeholder context was used instead of a proper one,
+    /// - Returns: Empty "to-do" baggage context which should be eventually replaced with a carried through one, or `background`.
+    public static func TODO(_ reason: StaticString? = "", function: String = #function, file: String = #file, line: UInt = #line) -> Baggage {
+        var context = Baggage.background
+        #if BAGGAGE_CRASH_TODOS
+        fatalError("BAGGAGE_CRASH_TODOS: at \(file):\(line) (function \(function)), reason: \(reason)")
+        #else
+        context[TODOKey.self] = .init(file: file, line: line)
+        return context
+        #endif
+    }
+
+    private enum TODOKey: BaggageKey {
+        typealias Value = TODOLocation
+        static var name: String? {
+            return "todo"
+        }
+    }
+}
+
+/// The `BaggageProtocol` should not directly be used in APIs.
+///
+/// Prefer accepting the `ContextProtocol` in APIs which need to accept a baggage context.
+/// If `Context` can not be used for some reason, it is preferable to accept a `Baggage` directly rather than this protocol.
+public protocol BaggageProtocol {
+    /// Get the `Baggage` container.
+    var baggage: Baggage { get set }
```

Comment:
`set` intentional?

Reply:
actually, I even wonder why the getter is part of the protocol that defines the baggage. its seems like this protocol is a mix of the baggage container(get maybe set) and the baggage itself (subscript, forEach). can you elaborate why we need to have them in one protocol?

Reply:
Discussed some more and dropping the BaggageProtocol entirely